src/Chapter2.hs

```diff
@@ -842,7 +867,12 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [Int] -> [Int]
+rotate times list
+  | mod times (length(list)) == 0 = list
+  | times > 0 = (take (length(list)) . drop (mod times (length(list)))) (cycle list)
+  | otherwise = []
```

Comment:
The implementation looks correct, but I see a few improvements:

* Check the list on emptiness first to avoid division by zero
* Check the `times` variable on invalid inputs (zero or negative)
* Move `mod times (length list)` into a separate variable (e.g. inside the `where`  clause) to avoid computing it twice
* Reduce the number of `()` to make the code more readable

Reply:
Great, I can see this be more efficient and I see how I can combine the case with where, removed not necessary parenthesis